src/sentry/api/endpoints/organization_searches.py

```diff
@@ -24,14 +25,42 @@ def get(self, request, organization):
         :auth: required
 
         """
-        org_searches = Q(
-            Q(owner=request.user) | Q(owner__isnull=True),
-            ~Q(query__in=DEFAULT_SAVED_SEARCH_QUERIES),
-            project__in=self.get_projects(request, organization),
-        )
-        global_searches = Q(is_global=True)
-        saved_searches = SavedSearch.objects.filter(
-            org_searches | global_searches
-        ).order_by('name', 'project')
-
-        return Response(serialize(list(saved_searches), request.user))
+        if request.GET.get('use_org_level') == '1':
+            org_searches_q = Q(
+                Q(owner=request.user) | Q(owner__isnull=True),
+                organization=organization,
+            )
+            global_searches_q = Q(is_global=True)
+            saved_searches = list(SavedSearch.objects.filter(
+                org_searches_q | global_searches_q,
+                type=SavedSearchType.ISSUE.value,
+            ).extra(
+                select={'has_owner': 'owner_id is not null'},
+                order_by=['-has_owner', 'name'],
+            ))
+            results = []
+            if saved_searches:
+                pinned_search = None
+                # If the saved search has an owner then it's the user's pinned
+                # search. The user can only have one pinned search.
+                results.append(saved_searches[0])
+                if saved_searches[0].owner is not None:
+                    pinned_search = saved_searches[0]
+                for saved_search in saved_searches[1:]:
+                    # If a search has the same query as the pinned search we
+                    # want to skip that search to avoid duplication.
+                    if pinned_search and saved_search.query == pinned_search.query:
+                        continue
+                    results.append(saved_search)
+        else:
+            org_searches = Q(
+                Q(owner=request.user) | Q(owner__isnull=True),
+                ~Q(query__in=DEFAULT_SAVED_SEARCH_QUERIES),
+                project__in=self.get_projects(request, organization),
```

Comment:
Should this also exclude saved searches that have `organization` set? I didn't think we wanted the two implementations to overlap data wise.

Reply:
There won't be any saved searches that have both `project` and `organization` set. I can guard against it if you're concerned though.

Reply:
It might not be necessary if we are careful in the put/post endpoint for organization searches.